pkg/api/runc.go

```diff
@@ -381,6 +382,10 @@ func (s *service) RuncQuery(ctx context.Context, args *task.RuncQueryArgs) (*tas
 		})
 	}
 
+	if len(containers) == 0 {
+		return nil, status.Error(codes.NotFound, "no containers found")
+	}
+
```

Comment:
```suggestion
```

Reply:
Note this change was required as in crioRootfsCheckpoint we do,

```go
			ctrResp, runcDiscoveryErr := cts.TaskService.RuncQuery(ctx, ctrByNameArgs)
			if runcDiscoveryErr == nil {
				runcId = ctrResp.Containers[0].ID
				break
			} else {
				logger.Error().Err(runcDiscoveryErr).Msg("error querying runc")
				fmt.Printf("Error: %s", runcDiscoveryErr)
			}
```

Note this is the actual fix here, as we unconditionally set runcId if there is not an error.